src/actions/requests.rs

```diff
@@ -81,8 +81,12 @@ impl RequestAction for WorkspaceSymbol {
             .limit(512);
         let defs = analysis.query_defs(query).unwrap_or_else(|_| vec![]);
 
+        // Sometimes analysis will return duplicate symbols for the same location,
+        // fix that up.
+        let mut seen_locations = HashSet::new();
         Ok(
             defs.into_iter()
+                .filter(|d| seen_locations.insert(d.span.clone()))
```

Comment:
You could use itertools here `.unique_by`. Might be nicer, but not a big deal.

Reply:
Oh, indeed, thanks!